packages/core/src/getPathFromState.tsx

```diff
@@ -0,0 +1,47 @@
+import { NavigationState, PartialState, Route } from './types';
+
+type State = NavigationState | Omit<PartialState<NavigationState>, 'stale'>;
+
+/**
+ * Utility to serialize a navigation state object to a path string.
+ * This is a lossy operation because we can't store all of the data in the path.
```

Comment:
This comment maybe is not needed? `Utility to serialize a navigation state object to a path string.` should be fine